src/main/java/quickcache/logic/parser/ParserUtil.java

```diff
@@ -164,6 +165,9 @@ public static Tag parseTag(String tag) throws ParseException {
      * @throws ParseException if the given {@code difficulty} is invalid.
      */
     public static Difficulty parseDifficulty(String difficulty) throws ParseException {
+        if (difficulty.isEmpty()) {
+            return new Difficulty(Difficulties.UNSPECIFIED.name());
+        }
         requireNonNull(difficulty);
```

Comment:
Think `requireNonNull(difficulty)` should come before `difficult.isEmpty()` since you can't call `isEmpty()` on `null`

Reply:
Ah yes thanks for pointing this out!